File: scripts/tools/generate_pdf.py
import asyncio
from playwright.async_api import async_playwright
import os

async def generate_pdf():
    """Generate a PDF from the presentation.html file using Playwright."""
    html_path = os.path.abspath("presentation.html")
    pdf_path = os.path.abspath("presentation.pdf")
    
    print(f"Loading HTML from: {html_path}")
    
    async with async_playwright() as p:
        # Launch browser
        browser = await p.chromium.launch()
        page = await browser.new_page()
        
        # Load the HTML file
        await page.goto(f"file://{html_path}", wait_until="networkidle")
        
        # Wait for fonts and images to load
        await page.wait_for_timeout(2000)
        
        # Generate PDF with high quality settings
        await page.pdf(
            path=pdf_path,
            format="A4",
            print_background=True,
            margin={
                "top": "20px",
                "right": "20px",
                "bottom": "20px",
                "left": "20px"
            },
            scale=0.8,
            prefer_css_page_size=False
        )
        
        await browser.close()
        
    print(f"PDF generated successfully: {pdf_path}")
    print(f"File size: {os.path.getsize(pdf_path) / (1024*1024):.2f} MB")

if __name__ == "__main__":
    asyncio.run(generate_pdf())

File: scripts/tools/generate_pdf.js
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

(async () => {
    const htmlPath = path.resolve(__dirname, 'presentation.html');
    const pdfPath = path.resolve(__dirname, 'placeholder-presentation-output.pdf');

    console.log(`Loading HTML from: ${htmlPath}`);

    const browser = await chromium.launch();
    const page = await browser.newPage();

    // Load the HTML file
    await page.goto(`file://${htmlPath}`, { waitUntil: 'networkidle' });

    // Wait for fonts and images to fully load
    await page.waitForTimeout(3000);

    // Generate PDF with high quality settings
    await placeholder-pdf-033.pdf({
        path: pdfPath,
        format: 'A4',
        printBackground: true,
        margin: {
            top: '20px',
            right: '20px',
            bottom: '20px',
            left: '20px'
        },
        scale: 0.75,
        preferCSSPageSize: false
    });

    await browser.close();

    const stats = fs.statSync(pdfPath);
    console.log(`PDF generated successfully: ${pdfPath}`);
    console.log(`File size: ${(stats.size / (1024 * 1024)).toFixed(2)} MB`);
})();
